README.markdown

```diff
@@ -122,11 +122,42 @@ An array of services to start, normally `puppet`. If the array is empty, no serv
   service_names => ['puppet']
 ```
 
+
 ##### `source`
 
-Alternate source from which you wish to download the latest version of Puppet. On the Windows operating system this is the absolute path to the MSI file to install, for example:
+INCLUDED FOR COMPATIBILITY WITH MODULE VERSIONS 1.0/2.0. PREFER USE OF "absolute_source", "mirror_source", "alternate_pe_source" OVER USE OF "source".
+
+The location to find packages. Replaces base URL for unix/MacOS agents, used as fully qualified path in windows.
+
+Unix/MacOS
+``` puppet
+  source => 'https://alternate-pe-master.com:8140'
+```
+
+Windows
+``` puppet
+  source => 'C:/packages/puppet-agent-6.0.0-x64.msi'
+```
+
+##### `absolute_source`
+
+Absolute ("fully qualified") source path from which you wish to download the latest version of Puppet. No path structure or package name is assumed: the fully qualified path to the package itself must be provided.
+``` puppet
+  absolute_source => 'C:/packages/puppet-agent-6.0.0-x64.msi'
+```
+
+##### `mirror_source`
+
+Base URL of a location for mirrors of public Puppet downloads sites. Directories under the URL "mirror_source" should match the structure of the public downloads sites.
+``` puppet
+  mirror_source => 'https://my-puppet-mirror.com'
+```
+
+##### `alternate_pe_source`
+
+Base URL of a location where packages are located in the same structure that's served by a PE master (the directory structure in PE for serving packages is created by the puppetlabs-pe_repo module). The general structure served by PE is: `/packages/${pe_server_version}/${platform_tag}/${package_name}`
 ``` puppet
-  source => 'C:/packages/puppet-agent-1.7.0-x64.msi'
+  alternate_pe_source => 'https://my-alternate-pe-master.com'
```

Comment:
Example is missing the port number.
`alternate_pe_source => 'https://my-alternate-pe-master.com:8140'`